jenkins/format-submission.py

```diff
@@ -0,0 +1,162 @@
+        jsonData = response.json()
+        for attachment in jsonData['fields']['attachment']:
+            filename = attachment['filename']
+            url = attachment['content']
+            response = requests.get(
+                url=url,
+                auth=AUTH
+            )
+            with open(filename, 'wb') as fout:
+                fout.write(response.content)
+            attachments.append(filename)
+    except requests.exceptions.RequestException as e:
+        raise SystemExit(e)
+    return attachments
+
+
+def get_issue():
+    try:
+        response = requests.get(
+            url='{}/rest/api/2/issue/{}'.format(HOST, ISSUE_ID),
+            headers=HEADERS,
+            auth=AUTH,
+        )
+        return response.json()
+    except requests.exceptions.RequestException as e:
+        raise SystemExit(e)
+
+
+def find_mapp_file(attachments):
+    index = 0
+    mapIndex = 0
+    mappNum = 0
+    for attachment in attachments:
+        if attachment.endswith('.mapp'):
+            mappNum += 1
+            mapIndex = index
+        if mappNum > 1:
+            _error_writeback("more than one .mapp file attached")
+        index += 1
+    if mappNum == 0:
+        _error_writeback("no .mapp file attached")
+    return attachments[mapIndex]
+
+
+def format_bundle(mappFile, privacyUrl, documentationUrl, termsOfUseUrl, supportUrl, vendor):
+    try:
+        with zipfile.ZipFile(mappFile, 'r') as zin:
+            # extract into temporary directory
+            tempDir = 'temp'
+            os.system('mkdir {}'.format(tempDir))
+            zin.extractall(tempDir)
+
+        with open(tempDir + '/' + METADATA_FILE, 'r+') as fin:
+            jsoncontents = json.load(fin)
+            jsoncontents['vendor'] = vendor
+            jsoncontents.pop('tags')
+            jsoncontents.update({'metadata': []})
+            jsoncontents['metadata'].append({'tag': 'privacyUrl', 'value': privacyUrl})
+            jsoncontents['metadata'].append({'tag': 'documentationUrl', 'value': documentationUrl})
+            jsoncontents['metadata'].append({'tag': 'termsOfUseUrl', 'value': termsOfUseUrl})
+            jsoncontents['metadata'].append({'tag': 'supportUrl', 'value': supportUrl})
+            
+            # create directory structure: ./http/<vendor>/<exportid> 
+            exportId = jsoncontents['id']
+            httpDir = 'http'
+            vendorDir = '{}/{}'.format(httpDir, vendor)
+            exportDir = '{}/{}/{}'.format(httpDir, vendor, exportId)
+            os.system('mkdir {}'.format(httpDir))
+            os.system('mkdir {}'.format(vendorDir))
+            os.system('mkdir {}'.format(exportDir))        
+            os.system('mv {}/* {}'.format(tempDir, exportDir))
+            os.system('mv {} {}'.format(mappFile, exportDir))    
+
+            fin.seek(0)
+            json.dump(jsoncontents, fin, indent=4)
+            fin.truncate()
+    except: 
+        _error_writeback("missing or malformed .mapp file")
```

Comment:
The writeback here should include the details about what was misformed, if we just include the python error message context it will do this for us automatically.

Reply:
Also the try block covers a large area of code which could fail for more reasons than just incorrect mapp file. For instance the directory creation, file rewrite, etc. If we want to verify the structure than we should do that explicitly by looking for the existence of the fields we expect to be there.